reporting.txt

```diff
@@ -0,0 +1,952 @@
+
+   o  _Applied policy_, consisting of:
+
+      *  One of the following policy types:
+
+         +  The SMTP MTA STS policy applied (as a string)
+
+         +  The DANE TLSA record applied (as a string)
+
+      *  The domain for which the policy is applied
+
+      *  The MX host
+
+      *  An identifier for the policy (where applicable)
+
+   o  _Aggregate counts_, comprising _result type_, _sending MTA IP_,
+      _receiving MTA hostname_, _message count_, and an optional
+      _additional information_ field containing a URI for recipients to
+      review further information on a failure type.
+
+   Note that the failure types are non-exclusive; an aggregate report
+   MAY contain overlapping "counts" of failure types where a single send
+   attempt encountered multiple errors.
+
+
+
+
+
+
+
+
+Margolis, et al.        Expires October 20, 2016                [Page 5]
+
+Internet-Draft                 SMTP-TLSRPT                    April 2016
+
+
+4.1.  Result Types
+
+   The list of result types will start with the minimal set below, and
+   is expected to grow over time based on real-world experience.  The
+   initial set is:
+
+4.1.1.  Success Type
+
+   o  "success": This indicates that the sending MTA was able to
+      successfully negotiate a policy-compliant TLS connection, and
+      serves to provide a "heartbeat" to receiving domains that
+      reporting is functional and tabulating correctly.
+
+4.1.2.  Routing Failures
+
+   o  "mx-mismatch": This indicates that the MX resolved for the
+      recipient domain did not match the MX constraint specified in the
+      policy.
+
+   o  "certificate-mismatch": This indicates that the certificate
+      presented by the receiving MX did not match the MX hostname
+
+4.1.3.  Negotiation Failures
+
+   o  "starttls-not-supported": This indicates that the recipient MX did
+      not support STARTTLS.
+
+   o  "invalid-certificate": This indicates that the certificate
+      presented by the receiving MX did not validate according to the
+      policy validation constraint.  (Either it was not signed by a
+      trusted CA or did not match the DANE TLSA record for the recipient
+      MX.)
+
+   o  "expired-certificate": This indicates that the certificate has
+      expired.
+
+4.1.3.1.  DANE-specific Policy Failures
+
+   o  "tlsa-invalid": This indicates a validation error for Policy
+      Domain specifying "tlsa" validation.
+
+   o  "dnssec-invalid": This indicates a failure to validate DNS records
+      for a Policy Domain with a published "tlsa" record.
+
+
+
+
+
+
+
+
+Margolis, et al.        Expires October 20, 2016                [Page 6]
+
+Internet-Draft                 SMTP-TLSRPT                    April 2016
+
+
+4.1.3.2.  STS-specific Policy Failures
+
+   o  "sts-invalid": This indicates a validation error for Policy Domain
+      specifying "STS" validation.
+
+   o  sender-does-not-support-validation-method: This indicates the
```

Comment:
I wonder if we still need to have this in here.  If you don't support the PKI validation, you're probably not going to query for STS records to begin with, correct?  Or you'd prefer that the sender explicitly state that even though they were successful in TLS negotiation (because without DANE/STS, any cert will qualify), they couldn't apply the STS that existed?